src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java

```diff
@@ -206,6 +207,15 @@ private ComponentIdentifier negotiateVersionWithCloud(String componentName,
                 resolvedComponentVersion = RetryUtils.runWithRetry(clientExceptionRetryConfig,
                         () -> componentServiceHelper.resolveComponentVersion(componentName, null, versionRequirements),
                         "resolve-component-version", logger);
+
+                VendorGuidance vendorGuidance = resolvedComponentVersion.vendorGuidance();
+                if (vendorGuidance != null && vendorGuidance.equals(VendorGuidance.DISCONTINUED)) {
```

Comment:
no need to nullcheck if you flip this around

`VendorGuidance.DISCONTINUED.equals(vendorGuidance)`

Reply:
huh, thats interesting